lib/i18n_js.rb

```diff
@@ -0,0 +1,26 @@
+def setup_i18n_js
+  environment(nil, env: 'development') do
+    <<~EOT
+      # Automatically generate the `translation.js` files
+      config.middleware.use I18n::JS::Middleware
+
+    EOT
+  end
+
+  insert_into_file 'app/assets/javascripts/application.js', after: "//= require activestorage\n" do
+    <<~EOT
+      //= require i18n
+      //= require translations/translations
+    EOT
+  end
+
+  append_to_file '.gitignore' do
+    <<~EOT
+      
+      # Ignore i18n.js generated files
+      /vendor/assets/javascripts/i18n.js
+      /app/assets/javascripts/translations/translations.js
```

Comment:
@rosle For this config, when we want to deploy the project to Heroku we can't push those files to Heroku.

Reply:
Added more comment regarding deployment on Heroku 🚀